ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegisterInjector.java

```diff
@@ -76,10 +78,15 @@ private void registerForCreateSource(final ConfiguredStatement<? extends CreateS
 
     final LogicalSchema schema = cs.getStatement().getElements().toLogicalSchema();
 
+    final KsqlConfig ksqlConfig = cs.getConfig()
+        .cloneWithPropertyOverwrite(cs.getConfigOverrides());
+
+    final KsqlTopic ksqlTopic = TopicFactory.create(cs.getStatement().getProperties(), ksqlConfig);
```

Comment:
SchemaRegisterInjector should delegate to `TopicFactory` to get key and value formats, so its not duplicating code.